aws/resource_aws_efs_file_system.go

```diff
@@ -76,6 +91,14 @@ func resourceAwsEfsFileSystemCreate(d *schema.ResourceData, meta interface{}) er
 		createOpts.PerformanceMode = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("encrypted"); ok {
+		createOpts.Encrypted = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("kms_key_id"); ok {
+		createOpts.KmsKeyId = aws.String(v.(string))
+	}
```

Comment:
We should check that `Encrypted` is true if `KmsKeyId` is set, and catch the related Exception if the API call is made (BadRequest perhaps?).